File: lista-de-exercicios/exercicio-14/ConversorHoras.ts
export class ConversorHoras {
    private tempo: number;

    public constructor (tempo: number) {
        this.tempo = tempo;
    }

    public getHoras(): string {
        return `${this.converterHoras()}h${this.receberMinutos()}min${this.receberSegundos()}seg`;
    }

    public receberSegundos() {
        const segundos = this.tempo % 60;
        return segundos;
    }

    public converterMinutos() {
        const minutos = Math.floor(this.tempo / 60);
        return minutos;
    }

    public receberMinutos() {
        const minutosSobra = this.converterMinutos() % 60;
        return minutosSobra;
    }

    public converterHoras() {
        const horas = Math.floor(this.converterMinutos() / 60);
        return horas;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio14/ConversorHoras.java
package org.example.exercicio14;

public class ConversorHoras {
    private int tempo;

    public ConversorHoras(int tempo) {
        this.tempo = tempo;
    }

    public String getHoras() {
        return this.converterHoras() + "h" + this.receberMinutos() + "min" + this.receberSegundos() + "seg";
    }

    public int receberSegundos() {
        int segundos = this.tempo % 60;
        return segundos;
    }

    public int converterMinutos() {
        int minutos = (int) Math.floor(this.tempo / 60);
        return minutos;
    }

    public int receberMinutos() {
        int minutosSobra = this.converterMinutos() % 60;
        return minutosSobra;
    }

    public int converterHoras() {
        int horas = (int) Math.floor(this.converterMinutos() / 60);
        return horas;
    }
}
